src/fsharp/IlxGen.fs

```diff
@@ -4969,7 +4969,8 @@ and GenMarshal cenv attribs =
                error(Error(FSComp.SR.ilCustomMarshallersCannotBeUsedInFSharp(),m))
                (* ILNativeType.Custom of bytes * string * string * bytes (* GUID,nativeTypeName,custMarshallerName,cookieString *) *)
                //ILNativeType.Error  
-            | 0x2D -> ILNativeType.Error  
+            | 0x2D -> ILNativeType.Error
+            | 0x30 -> ILNativeType.LPUTF8STR
             | _ -> ILNativeType.Empty
```

Comment:
Maybe we should throw a compile error here if we see invalid UnmanagedTypes?

Reply:
Yes, maybe. I've no idea why it doesn't do that already - or why it maps the original value through a couple of enums rather than just passing the original value along for that matter.